doc-deploy-changelog/action.yml

```diff
@@ -147,10 +156,21 @@ runs:
         with open(github_env, "a") as f:
             f.write(f"TOWNCRIER_DIR={directory}")
 
+    - name: "Get main branch name"
+      shell: bash
+      run: |
+        head_branch=$(git remote show origin | grep "HEAD branch:")
+        formatted_head_branch=${head_branch#*: }
+
+        echo "MAIN_BRANCH=$formatted_head_branch" >> $GITHUB_ENV
+
     - name: "Set CHANGELOG update branch"
       shell: bash
       run: |
-        if [[ ${{ inputs.update-release-branch }} == "true" ]]; then
+        if [[ ${{ inputs.update-main-branch }} == "true" ]]; then
+          echo "UPDATE_BRANCH=${{ env.MAIN_BRANCH }}" >> $GITHUB_ENV
+          echo "RELEASE_BRANCH_EXISTS=true" >> $GITHUB_ENV
+        elif [[ ${{ inputs.update-release-branch }} == "true" ]]; then
           # Assume release branch is "release/major.minor"
           # v0.1.0 (tag) -> release/0.1
```

Comment:
You are giving preference to one option over the other. IMO these options should be incompatible. You should verify (somewhere at the beginning) that only one of them is true. If all are false, it's not an issue I guess. But if both are true, that is a problem.